internal/service/alarms/internal/alarms_server.go

```diff
@@ -341,6 +346,107 @@ func (a *AlarmsServer) PatchAlarm(ctx context.Context, request api.PatchAlarmReq
 	return api.PatchAlarm200JSONResponse{AlarmAcknowledged: request.Body.AlarmAcknowledged, PerceivedSeverity: request.Body.PerceivedSeverity}, nil
 }
 
+// GetServiceConfiguration handles an API request to fetch the Alarm Service Configuration
+func (a *AlarmsServer) GetServiceConfiguration(ctx context.Context, _ api.GetServiceConfigurationRequestObject) (api.GetServiceConfigurationResponseObject, error) {
+	records, err := a.AlarmsRepository.GetServiceConfigurations(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Alarm Service Configuration: %w", err)
+	}
+
+	// There must always be a single record
+	if len(records) != 1 {
+		return nil, fmt.Errorf("expected a single Alarm Service Configuration record, but got %d", len(records))
+	}
+
+	object := models.ConvertServiceConfigurationToAPI(records[1])
+
+	return api.GetServiceConfiguration200JSONResponse(object), nil
+}
+
+// PatchAlarmServiceConfiguration handles an API request to patch the Alarm Service Configuration
+func (a *AlarmsServer) PatchAlarmServiceConfiguration(ctx context.Context, request api.PatchAlarmServiceConfigurationRequestObject) (api.PatchAlarmServiceConfigurationResponseObject, error) {
+	// Fetch the Alarm Service Configuration to be patched
+	records, err := a.AlarmsRepository.GetServiceConfigurations(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Alarm Service Configuration: %w", err)
+	}
+
+	// There must always be a single record
+	if len(records) != 1 {
+		return nil, fmt.Errorf("expected a single Alarm Service Configuration record, but got %d", len(records))
+	}
+
+	serviceConfigRecord := records[0]
+
+	// Patch the Alarm Service Configuration
+	if request.Body.RetentionPeriod != 0 {
```

Comment:
Perhaps we should make this a non-required field so that we can distinguish between it not being set at all by the client and being set below the minimum.

Reply:
It makes sense that it was nil, but I'm following the spec. The field is expected.